test/interfaces.ts

```diff
@@ -1,117 +1,133 @@
-import { Location, SearchAttributes } from './interfaces2';
```

Comment:
Please add tests for generics and nested types. If it is too out of the scope, maybe we can skip the tests

Reply:
There are already tests for nested types (interfaces extending from interfaces)

There are not for generics, I'll add some

Reply:
Added (disabled) tests for generics (since supporting generics will require additional refactors out of the scope)